src/Data/HeytingAlgebra.purs

```diff
@@ -1,9 +1,31 @@
 module Data.HeytingAlgebra
-  ( class HeytingAlgebra, tt, ff, implies, conj, disj, not
-  , (&&), (||)
+  ( class HeytingAlgebra
+
+  , tt
+  , ff
+  , implies
+  , conj
+  , disj
+  , not
+  , (&&)
+  , (||)
+
+  , class HeytingAlgebraRow
+  , ffRecordImpl
+  , ttRecordImpl
+  , impliesRecordImpl
+  , conjRecordImpl
+  , disjRecordImpl
+  , notRecordImpl
```

Comment:
Maybe we could drop `Impl` from all these kinds of names - usually that's for FFI things where there's a non-`Impl` version in scope too.

Perhaps the classes should be using `Record` in their name rather than `Row` too, since they are specialised to records in particular.

Reply:
Thinking about it, maybe all these should be defined in modules like `Data.HeytingAlgebra.Record` or something like that, that way they don't have to be exported from these "primary" modules, given that there's not really a need to use these functions outside of the record instances.

Reply:
Only problem here is that they all require things to have `HeytingAlgebra` constraints, which would cause a circular dependency :(

Reply:
In that case, how about putting all of this in `Data.HeytingAlgebra.Internal`, and then re-exporting the important bits from `Data.HeytingAlgebra`?